daemon/src/db.rs

```diff
@@ -13,14 +15,32 @@ pub fn setup(url: &str) -> Result<SqliteConnection> {
     info!("Using database at {:?}", url);
     let mut connection = SqliteConnection::establish(url)?;
 
+    let mut database_schema_changed = false;
     while connection
         .has_pending_migration(MIGRATIONS)
         .map_err(|err| anyhow!("Failed to check for pending migrations: {err:#}"))?
     {
-        let version = connection
-            .run_next_migration(MIGRATIONS)
+        let pending_migrations = connection
+            .pending_migrations(MIGRATIONS)
+            .map_err(|err| anyhow!("Failed to check for pending migrations: {err:#}"))?;
```

Comment:
I think this is checking twice and can be reduced to:
```rust
loop {
        let pending_migrations = connection
            .run_next_migration(MIGRATIONS)
            .pending_migrations(MIGRATIONS)
            .map_err(|err| anyhow!("Failed to check for pending migrations: {err:#}"))?;
```
The exit condition/instruction is done inside the loop. :)